test/integration/scheduler/podsready/scheduler_test.go

```diff
@@ -193,7 +193,7 @@ var _ = ginkgo.Describe("SchedulerWithWaitForPodsReady", func() {
 
 	var _ = ginkgo.Context("Short PodsReady timeout", func() {
 		ginkgo.BeforeEach(func() {
-			podsReadyTimeout = 3 * time.Second
+			podsReadyTimeout = 250 * time.Millisecond
```

Comment:
I decreased the timeout to 250 ms in all short podsready timeout cases.
HDYT? @mimowo @alculquicondor

Reply:
nice win! This will speed up the tests. Can we even go below that, say 10ms?

Reply:
I tried to use the 10ms, but test case the below failed:
https://github.com/kubernetes-sigs/kueue/blob/ff4ca7f566e4a12e13ffc3db62cbe896e7993685/test/integration/scheduler/podsready/scheduler_test.go#L261-L263

```shell
  [FAILED] Timed out after 25.000s.
  The function passed to Eventually failed at /Users/tenzen/go/src/sigs.k8s.io/kueue/test/util/util.go:354 with:
  Expected object to be comparable, diff:   &v1beta1.RequeueState{
  -     Count: &2,
  +     Count: &1,
        ... // 1 ignored field
    }
  
  In [It] at: /Users/tenzen/go/src/sigs.k8s.io/kueue/test/integration/scheduler/podsready/scheduler_test.go:261 @ 05/27/24 16:59:35.799
```

I guess that the workload controller could not observe the transition of the state.

Reply:
Hm, maybe this is because the scheduler admits the workload before we can set `active: false`. This might be the reason the time was set to 3s to make sure such race conditions don't happen. Probably reducing to 1s might be safer.

Also, for the new tests, maybe we could consider moving them here: https://github.com/kubernetes-sigs/kueue/blob/ff4ca7f566e4a12e13ffc3db62cbe896e7993685/test/integration/controller/jobs/job/job_controller_test.go#L1743C39-L1743C92. WDYT?

Reply:
> This guess doesn't seem to be correct since you added this case when you implemented the podsready feature here: https://github.com/kubernetes-sigs/kueue/commit/9d9336b9b54a646f4149e7ae717a566f1e40dd0c

makes sense, I don't remember why it was 3s, I'm ok to reduce it to as low value as possible, provided the tests continue to pass.

> I think that this case still valid to put here (scheduler+controller) since it would be better to keep observe not to happen the unexpected race conditions bwteen the scheduler and controllers.

Makes sense, on the other hand it exposes us to the race conditions. If you still think the tests are valuable with scheduler I'm ok to keep them here. Maybe we can consider two types of short timeouts: 10ms for tests which are not vulnerable to the race conditions, and longer 250ms or 1s for tests which are vulnerable?

Reply:
> Makes sense, on the other hand it exposes us to the race conditions. If you still think the tests are valuable with scheduler I'm ok to keep them here. Maybe we can consider two types of short timeouts: 10ms for tests which are not vulnerable to the race conditions, and longer 250ms or 1s for tests which are vulnerable?

That sounds reasonable, but we need to restructure podsready test. So, could we just increase 250ms to 1s in this PR, and then could we try to restructure podsready integration test in followup PR?

@mimowo WDYT?

Reply:
> That sounds reasonable, but we need to restructure podsready test. So, could we just increase 250ms to 1s in this PR, and then could we try to restructure podsready integration test in followup PR?

sgtm, thanks

Reply:
Done.
Also I opened an issue: https://github.com/kubernetes-sigs/kueue/issues/2285